src/main/java/com/puppycrawl/tools/checkstyle/gui/TreeTable.java

```diff
@@ -190,6 +192,64 @@ private void setColumnsInitialWidth() {
         getColumn("Type").setPreferredWidth(preferredTypeColumnWidth);
     }
 
+    /**
+     * Search node by Xpath.
+     *
+     * @param root {@code DetailAST} root ast element
+     * @param xpath {@code String} xpath query
+     * @param nodes {@code Deque<DetailAST>} stack of nodes in selection path
+     * @return {@code boolean} node found or not
+     */
+    private static boolean search(DetailAST root, String xpath, Deque<DetailAST> nodes) {
+        boolean result = false;
+        if (xpath.equals(XpathQueryGenerator.generateXpathQuery(root))) {
+            nodes.push(root);
+            result = true;
+        }
+        else {
+            if (root.hasChildren()) {
```

Comment:
This `if` is redundant. When the root has no children, `root.getFirstChild()` will return `null` and the entire loop will be skipped.

Reply:
@pbludov Done